src/plasmapy/particles/particle_class.py

```diff
@@ -1794,7 +1794,7 @@ def is_ion(self) -> bool:
         """
         return self.is_category("ion")
 
-    def ionize(self, n: int = 1, inplace: bool = False) -> Particle | None:
+    def ionize(self, n: int | float = 1, inplace: bool = False) -> Particle | None:  # noqa: PYI041
```

Comment:
```suggestion
    def ionize(self, n: int = 1, inplace: bool = False) -> Particle | None:
```
In nearly all cases, `n` should be an `int`, I think.